src/custom/components/AccountDetails/Transaction.tsx

```diff
@@ -337,10 +343,20 @@ function ActivitySummary(params: {
               <b>Limit price</b>
               <i>{orderSummary.limitPrice}</i>
             </SummaryInnerRow>
-            <SummaryInnerRow>
-              <b>Valid to</b>
-              <i className={isCancelled ? 'cancelled' : ''}>{orderSummary.validTo}</i>
-            </SummaryInnerRow>
+
+            {!orderSummary.fulfillmentTime && (
+              <SummaryInnerRow>
+                <b>Valid to</b>
+                <i className={isCancelled ? 'cancelled' : ''}>{orderSummary.validTo}</i>
+              </SummaryInnerRow>
+            )}
+
+            {orderSummary.fulfillmentTime && (
+              <SummaryInnerRow>
+                <b>Filled on</b>
+                <i>{orderSummary.fulfillmentTime}</i>
+              </SummaryInnerRow>
+            )}
```

Comment:
This is the kind of thing makes sense to have a ternary, like:

```typescript
            {orderSummary.fulfillmentTime ? (
              <SummaryInnerRow>
                <b>Filled on</b>
                <i>{orderSummary.fulfillmentTime}</i>
              </SummaryInnerRow>
            ) : ({
              <SummaryInnerRow>
                <b>Valid to</b>
                <i className={isCancelled ? 'cancelled' : ''}>{orderSummary.validTo}</i>
              </SummaryInnerRow>
            })}

Reply:
I thought about it, and looks like a good thing to do after all. Will re-factor.